components/odh-notebook-controller/controllers/notebook_controller.go

```diff
@@ -77,6 +78,8 @@ type OpenshiftNotebookReconciler struct {
 // +kubebuilder:rbac:groups=rbac.authorization.k8s.io,resources=roles,verbs=get;list;watch;create;update;patch
 // +kubebuilder:rbac:groups=rbac.authorization.k8s.io,resources=rolebindings,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups="image.openshift.io",resources=imagestreams,verbs=list;get;watch
+// +kubebuilder:rbac:groups="datasciencepipelinesapplications.opendatahub.io",resources=datasciencepipelinesapplications,verbs=get;list;watch
+// +kubebuilder:rbac:groups="components.platform.opendatahub.io",resources=dashboards,verbs=get;list;watch
```

Comment:
Shouldn't we also have watches for datasciencepipelineapplication and dashboard, same as we have watch for configmaps in

https://github.com/opendatahub-io/kubeflow/blob/4af55e7b31faed44611fafc91d90907a2e784da0/components/odh-notebook-controller/controllers/notebook_controller.go#L484-L489
?

Reply:
A DSPA or Dashboard change is only relevant when the Notebook is started or re-started, because the Secret is mounted once at pod creation. The reconcile loop already runs on every start/update event, so adding additional watches on DSPA or Dashboard would only enqueue reconciles that cannot affect the running pod. Therefore I left them out to avoid unnecessary wakes.